### DISCUSSION TITLE
Hyperparameters optimization

### DISCUSSION THREAD

**Discussion by nikitaqwerty** [google/lightweight_mmm#110] [Ideas]
Hey there!

I successfully used lightweight_mmm for my company needs. In my specific case I've found very useful to optimize hyperparameters of the model using [hyperopt](https://github.com/hyperopt/hyperopt). In my case it saved me a lot of time to choice right model, to find better prior distribution parameters and brought me to meaningfully better results on validation set rather than whole manual parameter tuning.

However it took me some time of coding and debugging to create a reliable hyperparameter optimization and validation pipeline for this specific model.

I have some raw prototypes for this feature, but it still needs some coding work to be ready to get added as a feature to lightweight_mmm library. If community will find it useful I'll be glad to finish this work, refactor my code and contribute it to this repository as a new feature.

Thanks

**Comment by rohansinghh544** [comment]
can you please share your code for the same.

I am also working on similar project for my company. It would be really helpful if I can save some time using your code.

**Comment by AruunnAnanda** [comment]
Thanks for taking the initiative to ask. 

It would be of great help to the community if you can share the code when you finish it. 

I would also love to learn how you iterated and reached a step where you managed to get a working model. Here's my [LinkedIn ID](https://www.linkedin.com/in/arun-ananda/) if you would like to share your experience. Thanks!
